tiledb/sm/array_schema/enumeration.h

```diff
@@ -36,6 +36,8 @@
 #include <iostream>
 
 #include "tiledb/common/common.h"
+#include "tiledb/common/memory_tracker.h"
```

Comment:
Don't pull memory_tracker.h into this header. Use a forward declaration for MemoryTracker instead.